### DISCUSSION TITLE
Bitwarden Directory Connector - Failed to fetch

### DISCUSSION THREAD

**Discussion by EthanAmvt** [dani-garcia/vaultwarden#4602] [Q&A]
Hello,

I am a student currently interning at a company. I am working on a project to implement a password manager for the company. Specifically, I need to deploy a Vaultwarden manager at an enterprise scale.

Following your advice, I downloaded the Bitwarden Directory Connector application. However, despite its compatibility with Vaultwarden, I encountered the following error: "An error has occurred. Failed to fetch." I have filled in the three fields in Settings.

Do you have any solutions to this problem? I couldn't find anything online.

Thank you.
![pblm2](https://github.com/dani-garcia/vaultwarden/assets/170437404/6d73f95e-a8aa-4f92-84e8-0ceabb475ff3)
![pblm1](https://github.com/dani-garcia/vaultwarden/assets/170437404/057f3af1-c29e-4e31-bec1-58c890b114df)

**Comment by BlackDex** [comment]
Only fill in the first field. `Server URL` and leave the rest blank.

**Reply by EthanAmvt** [reply]
I put the vaultwarden website URL and I have the same error

**Reply by BlackDex** [reply]
The try to run it via the command-line and see if you get any errors there which might help.

**Reply by EthanAmvt** [reply]
Thank you for the advice, but I'm not sure what you are referring to. Could you tell me where I can find the commands to execute it?

**Reply by jobritz** [reply]
The Bitwarden Directory Connector also has an CLI Version https://bitwarden.com/help/directory-sync-cli/

**Reply by BlackDex** [reply]
If you run the desktop executable via a terminal, it should also output some error messages.

**Comment by EthanAmvt** [comment]
the issue is still relevant, if anyone has the answer I'm interested

**Reply by jobritz** [reply]
Have you tried to log in via the CLI as mentioned before?

**Reply by EthanAmvt** [reply]
I can't try with the CLI at the moment. After rereading my problem, I think it's because my vaultwarden manager runs on a private IP and is not published on the Internet. Could this be it?

**Reply by jobritz** [reply]
I don't think so, i'm running Vaultwarden withour any internet access and it works fine, even with the connector. I'm smelling a certificate issue but it would be nice to see what the CLI says. Do you use self-signed certificates?

**Reply by EthanAmvt** [reply]
Yes, I am using self-signed certificates, and my Vaultwarden connection is over HTTPS. However, when I try to run bwdc.exe (the CLI, I guess?), a terminal window opens and then shuts down instantly.

**Reply by BlackDex** [reply]
you need to run it via a terminal instead of running the exe via the file explorer.
Try using something like [Windows Terminal](https://www.microsoft.com/store/productId/9N0DX20HK701?ocid=pdpshare)

**Reply by jobritz** [reply]
> Yes, I am using self-signed certificates, and my Vaultwarden connection is over HTTPS. However, when I try to run bwdc.exe (the CLI, I guess?), a terminal window opens and then shuts down instantly.

I wouldn't use self-signed certificates. If u added them to your trusted certs it should be fine but if you don't this might be your issue

**Comment by DaveSH12** [comment]
Sorry for answering 1,5 years later. We had the same Problem and our Solution was using an older Version of BWDC. For us Bitwarden Directory Connector for Windows Version 2025.7.0 is working. With any newer Version there is the "Failed to fetch" Error and the URL could not be resolved. So if you still have this error, use another Version ;-)

**Comment by bauerstefan** [comment]
Problem still persists in latest version 2026.6.1-x86_64.AppImage
```
08:16:26.075 › TypeError: Failed to fetch
    at NodeApiService.nativeFetch (file:///tmp/.mount_BitwarvAfrtD/resources/app.asar/app/main.js:7044:16)
    at NodeApiService.fetch (file:///tmp/.mount_BitwarvAfrtD/resources/app.asar/app/main.js:6869:21)
    at async NodeApiService.postIdentityToken (file:///tmp/.mount_BitwarvAfrtD/resources/app.asar/app/main.js:6821:26)
    at async auth_service_AuthService.logIn (file:///tmp/.mount_BitwarvAfrtD/resources/app.asar/app/main.js:4474:26)
08:16:26.085 › Received API exception: TypeError: Failed to fetch

```
